lib/player.js

```diff
@@ -5947,6 +5947,42 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     return this.manifest_;
   }
 
+  /**
+   * Gets information about the currently playing video, audio, and text.
+   * In the case of a multi-codec or multi-mimeType manifest, this can let you
+   * determine the exact codecs and mimeTypes being played at the moment.
+   * @return {!shaka.extern.PlaybackInfo}
+   */
+  getCurrentPlaybackInfo() {
+    const variant = this.streamingEngine_.getCurrentVariant();
```

Comment:
This is going to get the variant that is currently being fetched, which may be different from the variant that is being used at the playhead.

Reply:
We don't currently track that.  To solve your problem, it seems like we would need to do so first.

If we did, we would have to be careful not to create memory leaks, e.g. a timeline that maps time ranges to variants, but variant objects get recreated on every manifest refresh, and the timeline still holds references to old ones that now can't be GC'd....

Reply:
note that code added to report quality change events captures this information. See quality_observer

Reply:
@caridley Quality observer now works for MSS and HLS, but it has a defect and that is that it only works with MP4, if TS, TTML or WebVTT is used (without MP4), as there is no initialization segment QualityObserver does not work.

Reply:
Would it work if ts, ttml, and webvtt include init segments? While uncommon, it is possible.

I wonder if it's easy to know whether there are init segments or not and if there aren't then use regular segments to inform the quality observer? Could specifically look to see if it was appended after a switch.

Reply:
For now I am removing the `getCurrentPlaybackInfo` function, and just integrating the bandwidth recording part with the `getFetchedPlaybackInfo` function that Alvaro made. Perhaps `getCurrentPlaybackInfo` can come in a future PR, if it looks like it's still necessary.